Android studio template

Ffloriel opened the pull request:
https://github.com/apl-devs/AppIntro/issues/440

avluis commented:
I'm seeing a few things that need to be reworked/rewritten (mostly wording, etc), but nothing major.
I'll try and get a review done as soon as I'm able (fairly busy).

Ffloriel commented:
No problem. Take your time.
The template is also a good opportunity to give an opinion about how to implement the library.
If you want to enforce the mCamelCase, declare the variables needed as properties or in the OnCreate function, etc...

paolorotolo commented:
Awesome!
@avluis So can I assign you the review of this PR?

avluis commented:
@paolorotolo Yep~ Go right ahead~

avluis commented on template/Android Studio 2/README.md:
```diff
@@ -0,0 +1,12 @@
+It also create strings arrays for the titles and the descriptions of the slides.
+The template updates the Android Manifest to add the class.
+
+![Template App Intro in action](./template_app_intro_visual.gif "Template App Intro in action")
```
@Ffloriel Can you go ahead and update the link to this image file?

avluis commented on template/Android Studio 2/recipe.xml.ftl:
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<recipe>
+
+    <dependency mavenUrl="com.github.paolorotolo:appintro:4.1.0" />
```
This may need to be edited as we are now using JitPack.io for future releases:
https://github.com/apl-devs/AppIntro/releases/tag/v4.2.0

paolorotolo commented:
Awesome @Ffloriel, we'll merge as soon as you push the chnages.

gNiKnet commented:
Just wanted to let you know that your demo app for AppIntro is not
completely working. On my Android 6.0.1 the whole app is blank. There are
no buttons nor text to test the examples.

avluis commented:
@gNiKnet There is already an issue for that, so if you find out anything more, go ahead and continue that conversation in there (tag @Andrew-Quebe as well): https://github.com/apl-devs/AppIntro/issues/442

Ffloriel commented:
Sorry for the wait.  
I don't know how to add jitPack repository with the template yet. So I added a comment at the top of the class to add jitPack repository. I also added it on the README.

I deleted the READMEs inside Android Studio folders. I think it is not necessary since there is one at the root of the template folder.

avluis commented:
All is looking good so far -- I have a side project to get out of the way first then I'll be back for a second look (possibly merge).
Though, just to get this out of the way now --
@Ffloriel Can we get rid of the archive requirement?
Basically, have the user download the [latest](https://github.com/apl-devs/AppIntro/releases/latest) ([zip](https://github.com/apl-devs/AppIntro/archive/master.zip)/[tar](https://github.com/apl-devs/AppIntro/archive/master.tar.gz) of the) repo intead of bundling the template in there, and then have them decompress the appropiate files.

While it does make it redundant on the user's end -- it will make it easier on our end when pushing updates (the archives would need to be recreated each time).
What do you think?

Ffloriel commented:
I thought of that. I decided to put the archive files from my experience wanting to get a folder from a repo and ended up downloading the whole project. 
While I think it is not the best from an user perspective, **it make sense to remove the archives**. The templates are in the template folder at the root of the project where they can easily find them. 
The user will have to extract an archive either way.

paolorotolo commented:
@avluis are you ok to merge now?

paolorotolo commented:
Alright, thanks for the contribution @Ffloriel
